### ISSUE TITLE
cbrtl() is not being built, and POSIX conformance

### ISSUE THREAD

**Issue by EdSchouten** [JuliaLang/openlibm#79]
It seems that s_cbrtl.c is not added to the list of source files in src/Make.files. It's explicitly commented out. Is this function still incomplete? If so, should we add a comment there? If not, shall we hook it up to the build?

**Comment by ViralBShah** [comment]
I recollect having some issues with it earlier, which may no longer hold. If it builds, we should hook it up again.

**Comment by EdSchouten** [comment]
Hi Viral,

It seems to build. I'll hook it up.

By the way, what are your thoughts in general on functions that are not part of the C standard/POSIX? In particular:

- gamma() (a misspelling of lgamma),
- finite() (a misspelling of isfinite),
- scalb() (a misspelling of scalbn),
- ...

I'd argue that these functions would normally be good to preserve if you have an existing commonly used implementation (FreeBSD's C library), but it would make little sense to provide those in newer or standalone implementations. I have these functions disabled in my local copy of OpenLibm, for example.

Maybe we could consider just removing them from the codebase and focussing on providing just the C/POSIX bits? If you want, I can do that at the same time.

**Comment by ViralBShah** [comment]
How about having an `ifdef` so that the default is POSIX, but one can build with `-DLIBM_EXTENSIONS`? 

I think we may be using some of these in Julia. I certainly would want us to use the POSIX compliant bits only, but that would mean changing a few things on the julia side. I think POSIX compliance is the right thing to do - although I should confess that I don't know what all is part of the standard and what is not.

Once this is done, we will review the Julia usage, and remove the extensions.

**Comment by ViralBShah** [comment]
BTW, if you know of any other libm test suites, they will be great to have.

**Comment by EdSchouten** [comment]
Something like -DLIBM_EXTENSIONS sounds all right. I'll see what I can do.

**Comment by ViralBShah** [comment]
@EdSchouten You are probably busy with other things, but would be great to have a PR to make the library POSIX compliant. Coming to think of it, we can avoid LIBM_EXTENSIONS altogether. I will fix whatever is needed on the Julia side, after the cbrtl and POSIX conformance changes, if you can provide those.

I am thinking of releasing openlibm 0.5 after these. Is there anything else we should do as part of the 0.5 release?

**Comment by nalimilan** [comment]
@ViralBShah As I said, I think we should do something about https://github.com/JuliaLang/openlibm/issues/55 before a new release.

**Comment by EdSchouten** [comment]
Hi there!

Thought I'd better respond. Yes! I'll make sure to send you a pull-request for this. I'll hopefully manage to do this some time upcoming weekend, as I'm currently busy putting a laminated floor in my new apartment. ;-)

Ed

**Comment by ViralBShah** [comment]
Closed by #85.
